src/Features/Core/Portable/Completion/CommonCompletionProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal abstract class CommonCompletionProvider : CompletionProvider
+    {
+        public override bool ShouldTriggerCompletion(SourceText text, int position, CompletionTrigger trigger, OptionSet options)
+        {
+            switch (trigger.Kind)
+            {
+                case CompletionTriggerKind.Insertion:
+                    var insertedCharacterPosition = position - 1;
```

Comment:
This seems to be undoing the +1 you do earlier.  Why do we need to do either hte +1 or hte -1?